From 0e7ef220d21fb127dd1dd2ec92fbffa40e2d665d Mon Sep 17 00:00:00 2001
From: Wojciech Bartczak <wbartczak@marvell.com>
Date: Wed, 11 Aug 2021 10:35:26 -0700
Subject: [PATCH 1900/1921] driver: soc: Modify error reporting for some SoC
 drivers

The change modifies error reporting for CN10KX SoC drivers.
Warnings for non-supported platforms have been replaced
with debug information displayed in dmesg.

Signed-off-by: Wojciech Bartczak <wbartczak@marvell.com>
Change-Id: Ic6067220f333f1166e995f2c67b3018f16454de6
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/58845
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/cn10k-rpram.c           | 2 +-
 drivers/soc/marvell/cn10k_serdes_diag.c     | 2 +-
 drivers/soc/marvell/cn10ka-swup/mrvl_swup.c | 2 +-
 drivers/soc/marvell/marvell_mac_mgmt.c      | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/marvell/cn10k-rpram.c b/drivers/soc/marvell/cn10k-rpram.c
index 5828989d9829..37f30426862d 100644
--- a/drivers/soc/marvell/cn10k-rpram.c
+++ b/drivers/soc/marvell/cn10k-rpram.c
@@ -149,7 +149,7 @@ static int __init cn10k_rpram_init(void)
 	ret = octeontx_soc_check_smc();
 
 	if (ret != 2) {
-		pr_info("%s: Not supported\n", __func__);
+		pr_debug("%s: Not supported\n", __func__);
 		return -EPERM;
 	}
 
diff --git a/drivers/soc/marvell/cn10k_serdes_diag.c b/drivers/soc/marvell/cn10k_serdes_diag.c
index 2a68f0870581..d3d743737405 100644
--- a/drivers/soc/marvell/cn10k_serdes_diag.c
+++ b/drivers/soc/marvell/cn10k_serdes_diag.c
@@ -922,7 +922,7 @@ static int __init serdes_dbg_init(void)
 	struct arm_smccc_res res;
 
 	if (octeontx_soc_check_smc() != 2) {
-		pr_info(DRV_NAME": Not supported\n");
+		pr_debug(DRV_NAME": Not supported\n");
 		return -EPERM;
 	}
 
diff --git a/drivers/soc/marvell/cn10ka-swup/mrvl_swup.c b/drivers/soc/marvell/cn10ka-swup/mrvl_swup.c
index d967816d50c9..a39cbf6a3acd 100644
--- a/drivers/soc/marvell/cn10ka-swup/mrvl_swup.c
+++ b/drivers/soc/marvell/cn10ka-swup/mrvl_swup.c
@@ -478,7 +478,7 @@ static int __init mrvl_swup_init(void)
 
 	ret = octeontx_soc_check_smc();
 	if (ret != 2) {
-		pr_info("%s: Not supported\n", __func__);
+		pr_debug("%s: Not supported\n", __func__);
 		return -EPERM;
 	}
 
diff --git a/drivers/soc/marvell/marvell_mac_mgmt.c b/drivers/soc/marvell/marvell_mac_mgmt.c
index 32b09aa94aab..5d38f3dd4ad3 100644
--- a/drivers/soc/marvell/marvell_mac_mgmt.c
+++ b/drivers/soc/marvell/marvell_mac_mgmt.c
@@ -222,7 +222,7 @@ static int __init cn10k_mac_mgmt_init(void)
 
 	ret = octeontx_soc_check_smc();
 	if (ret != 2) {
-		pr_info("%s: Not supported\n", __func__);
+		pr_debug("%s: Not supported\n", __func__);
 		return -EPERM;
 	}
 
-- 
2.31.1

